main/lsp/json_types.cc

```diff
@@ -175,4 +175,31 @@ unique_ptr<Diagnostic> Diagnostic::copy() const {
     return d;
 }
 
+string TextDocumentContentChangeEvent::apply(string oldSource) const {
```

Comment:
Observation:  could this take a `string_view`, and use the copy on line 187 in line 191, so that a `string_view` can avoid the allocation if it takes the `else` side of this method?

Reply:
Good question. I can make that change. (I actually already avoid the allocation with `move`, but `string_view` is cleaner.)